tests/foreman/smoke/test_ui_smoke.py

```diff
@@ -365,3 +370,141 @@ def test_end_to_end(self):
                 # Verify if package is installed by query it
                 result = vm.run(u'rpm -q {0}'.format(package_name))
                 self.assertEqual(result.return_code, 0)
+
+    def test_puppet_install(self):
+        """@Test: Perform puppet end to end smoke tests using RH repos.
+
+        @Feature: Smoke test puppet install and configure on client
+
+        @Assert: Client should get configured by puppet-module.
+
+        """
+        activation_key_name = gen_string('alpha', 6)
+        cloned_manifest_path = manifests.clone()
+        cv_name = gen_string('alpha', 6)
+        env_name = gen_string('alpha', 6)
+        org_name = gen_string('alpha', 6)
+        product_name = gen_string('alpha')
+        puppet_module = 'motd'
+        puppet_repository_name = gen_string('alpha')
+        repos = self.sync.create_repos_tree(SAT6_TOOLS_TREE)
+        rhel_prd = 'Employee SKU'
+        rhel6_repo = conf.properties['clients.rhel6_repo']
```

Comment:
Maybe should be better to create a helper for this, get_rhel6_repo. But we can refactor it later.